pkg/conditions/conditions.go

```diff
@@ -110,11 +112,33 @@ func ReconcileAgentFailed(err error) *ErrorCondition {
 func ReconcileConsolePluginFailed(err error) *ErrorCondition {
 	return &ErrorCondition{
 		Condition: metav1.Condition{
-			Type:    TypeReady,
-			Status:  metav1.ConditionFalse,
+			Type:    TypeFailed,
 			Reason:  "ReconcileConsolePluginFailed",
 			Message: "Failed to reconcile Console plugin: " + err.Error(),
 		},
 		Error: err,
 	}
 }
+
+// set previous conditions to false as FlowCollector manage only one status at a time
+func clearPreviousConditions(fc *flowslatest.FlowCollector) {
+	for _, existingCondition := range fc.Status.Conditions {
+		existingCondition.Status = metav1.ConditionFalse
+		meta.SetStatusCondition(&fc.Status.Conditions, existingCondition)
+	}
+}
+
+// sort conditions by date, latest first
+func sortConditions(fc *flowslatest.FlowCollector) {
+	sort.SliceStable(fc.Status.Conditions, func(i, j int) bool {
+		return !fc.Status.Conditions[i].LastTransitionTime.Before(&fc.Status.Conditions[j].LastTransitionTime)
+	})
+}
+
+// add a single condition to true, keeping the others to status false
+func AddUniqueCondition(cond *metav1.Condition, fc *flowslatest.FlowCollector) {
+	clearPreviousConditions(fc)
+	cond.Status = metav1.ConditionTrue
+	meta.SetStatusCondition(&fc.Status.Conditions, *cond)
+	sortConditions(fc)
+}
```

Comment:
while you are here lets add ut for condition package

Reply:
Sure I'll do that 👌

Reply:
I've added some tests but these seems flaky :thinking: 
https://github.com/netobserv/network-observability-operator/pull/403/commits/fda8f2adb06a16d743752b232c9e9cb3d833315a

I'm digging to find why

Reply:
I cannot recall now, maybe this UpdateCR isn't necessary, but I guess if I added that it's because the kube test env didn't trigger the reconcile as I expected (the goal here being to make sure the ServiceMonitor is recreated - I remember I added this test after a bug where SM wasn't reconciled.

I remember that a while ago [I removed](https://github.com/netobserv/network-observability-operator/commit/ca61fad47cf624b253d0b29ac9ebba8b3e218f69#diff-363292a59bf24b26bb1b9f7996bb46014a7c4b7cb2bb483633eef84cf20ba934R799) some condition-checking code in tests because iirc it wasn't working as expected with kube testenv, and led to flaky tests ... but I don't remember all the details